stackstorm-compose/st2web-healthcheck.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+# Check each service through the nginx reverse proxy for a specific return code. If the curl request
+# fails to work through nginx, a stop signal will be sent to nginx, causing the container to restart.
+API_STATUS=$(curl --write-out "%{http_code}\n" --silent --output /dev/null http://localhost/api/v1/)
+if [ "${API_STATUS}" != "401" ]; then  echo "st2api failure"; nginx -s stop; fi
```

Comment:
@ytjohn restarting st2web here really delays the initial boot duration, like 30 seconds to a minute. Do we really need that? I've tried some basic tests and it didn't hurt removing it.

Reply:
This health check was added covers the case of st2api's IP changing if st2api is [restarted while st2web is up](https://github.com/StackStorm/st2-dockerfiles/pull/28#issuecomment-626248413). If instead of `nginx- s stop` you do an `exit 1`, the container is marked as unhealthy.  However, docker-compose offers no facility to restart an unhealthy container. Stopping nginx kills the container, which (due to the restart on-failure) gets the container to restart.

I think what is happening is that In your testing, st2api fails at least once during startup because it can't reach rabbitmq. Without the health check causing an exit, st2web would be up and operational waiting on st2api to come back up. If st2api didn't fail at startup, then st2web would not fail.

There may be a way to improve on this. We're checking through http://localhost/st2api to make sure nginx has the right dns. The OS itself should always have the right dns. If we can check http://st2api/api/v1 directly earlier in the script, with an immediate exit 1, then we can cover both cases.  Case 1 is just to be unhealthy while downstream services are unavailable; Case 2 is that nginx's dns is stale and st2web needs restarted.

Can you try adding something like this at near the top of your st2wweb-healthcheck.sh, as the first check and see if it makes it better? Try and watch `docker-compose ps` to see the health of st2web and st2api during the initial startup.

```
DIRECT_API_STATUS=$(curl --write-out "%{http_code}\n" --silent --output /dev/null http://st2api/api/v1/)
if [ "${DIRECT_API_STATUS}" != "401" ]; then  echo "st2api unreachable"; exit 1; fi
```

Reply:
I pushed a new health check, which should take care of this.